src/Homework/Week3/Pb2/Customer.java

```diff
@@ -0,0 +1,58 @@
+package Homework.Week3.Pb2;
+
+import java.util.List;
+
+public class Customer {
+    private String name;
+    private MemberShip myMemberShip;
+    {
+        myMemberShip = MemberShip.NONE;
+    }
+    private Basket myBasket;
+
+    Customer(String name, MemberShip memberShip)
+    {
+        myMemberShip = memberShip;
+        myBasket = new Basket();
+    }
+
+    void addToBasket(ItemForSale item) {
+        if(item == null) {
+            System.out.println("The item added to the basket is null");
+            return;
+        }
+        myBasket.addItem(item);
+    }
+
+    float basketPrice()
+    {
+        if(myMemberShip.equals(MemberShip.NONE))
+            return myBasket.getTotalPrice();
+        else if(myMemberShip.equals(MemberShip.SILVER))
+            return myBasket.getTotalPrice()*90/100;
+        else if(myMemberShip.equals(MemberShip.GOLD))
+            return myBasket.getTotalPrice()*80/100;
+        return 0;
```

Comment:
cazul asta nu ar trebui sa existe, nu? deci 0 nu e o valoare potrivita, mai degraba ar trebui sa arunci o exceptie, in care sa pui un mesaj sugestiv, cum ca membershipul e invalid si astfel pe viitor, daca in enum vei adauga inca un tip de membership, vei vedea prin codul tau exceptia asta de membership invalid si vei putea sa vii si sa adugi implementarea pentru noul membership